### PR TITLE
drivers/periph: flashpage: add common helper functions

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

`flashpage_addr()` is a generic function that can be used by other implementations as well.

### Testing procedure

No code changes, everything should still compile.

### CODE REVIEW

**Comment by fjmolinas** [comment]
> flashpage_addr() is a generic function that can be used by other implementations as well.

But is there any? AFAIK stm32f2/4/7 are the only boards with nonuniform pages.

**Comment by maribu** [comment]
> But is there any? AFAIK stm32f2/4/7 are the only boards with nonuniform pages.

E.g. the lpc23xx MCUs also have non-uniform flash page sizes. Maybe @benpicco wants to implement `periph_flashpage` for them?

File: cpu/stm32/include/cpu_conf.h [hunk 1]
```diff
@@ -123,6 +123,7 @@ extern "C" {
 #if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
     defined(CPU_FAM_STM32F7)
 #define PERIPH_FLASHPAGE_CUSTOM_PAGESIZES
+#define PERIPH_FLASHPAGE_NEEDS_FLASHPAGE_ADDR
```
Comment (bergzand) [inline cpu/stm32/include/cpu_conf.h:126]: Mind adding some doxygen for this to `drivers/include/periph/flashpage.h`? Similar to how it is done for `PERIPH_FLASHPAGE_CUSTOM_PAGESIZES`

Comment (benpicco) [reply cpu/stm32/include/cpu_conf.h:126]: added

**Comment by fjmolinas** [comment]
@bergzand your comments have been addressed

**Review by MrKevinWeiss — APPROVED** [review]
Looks good, ACK.  Will await future cpus being cleaned up!

**Comment by MrKevinWeiss** [comment]
Waiting for some hil results (https://ci.riot-os.org/hil/blue/organizations/jenkins/RIOT_tests/detail/RIOT_tests/8/pipeline) but should be good.

**Review by MrKevinWeiss — CHANGES_REQUESTED** [review]
Blocking while figure out why the hil is reporting failures on the `nucleo-f091rc`, `nucleo-f103rb` and `nucleo-f303re`.

This could just be due to rebasing on master.  [Rerunning on the PR](https://ci.riot-os.org/hil/blue/organizations/jenkins/RIOT_tests/detail/RIOT_tests/10/pipeline/)

**Comment by benpicco** [comment]
rebased onto master

**Comment by MrKevinWeiss** [comment]
It seems there is a queue.  [Lets see if the rebase fixed it](https://ci.riot-os.org/hil/blue/organizations/jenkins/RIOT_tests/detail/RIOT_tests/15/pipeline).

**Review by MrKevinWeiss — APPROVED** [review]
ACK (again). It seems the rebase fixed the tests.  That functionality will have to be added to the HiL

**Pull request merged**